packages/ui/src/views/repo/components/branch-selector/branch-selector.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button, DropdownMenu, DropdownMenuTrigger, Icon, Text } from '@/components'
+import { cn } from '@utils/cn'
+
+import { BranchSelectorDropdown, type BranchSelectorDropdownProps } from './branch-selector-dropdown'
+
+interface BranchSelectorProps extends BranchSelectorDropdownProps {
+  size?: 'default' | 'sm'
+  prefix?: string
+  className?: string
+}
+
+export const BranchSelector: React.FC<BranchSelectorProps> = ({
+  name,
+  branchList,
+  tagList,
+  size = 'default',
+  selectBranch,
+  prefix,
+  className,
+  repoId,
+  spaceId
+}) => {
+  const isTag = tagList.some(tag => tag.name === name) ?? false
```

Comment:
Yes, you’re right. I’ve left a TODO in the code to revisit this later and improve the validation logic.

Reply:
Unfortunately, in this case, it will throw an error due to the double negation.
<img width="493" alt="Cursor 2024-11-28 23 17 06" src="https://github.com/user-attachments/assets/34f0dca9-492d-4241-95f7-3177581e3ed9">

Therefore, I think we can simplify it even further to just `files?.length`